File: leetcode/3314. Construct the Minimum Bitwise Array I/main.kt
class Solution {
    fun minBitwiseArray(nums: List<Int>): IntArray {
        return IntArray(nums.size) { idx ->
            val value = nums[idx]
            var result = -1
            
            for (i in 0 until value) {
                if (i or (i + 1) == value) {
                    result = i
                    break
                }
            }

            result
        }
    }
}

File: leetcode/3314. Construct the Minimum Bitwise Array I/main.rs
impl Solution {
    pub fn min_bitwise_array(nums: Vec<i32>) -> Vec<i32> {
        let mut result = vec![-1; nums.len()];

        for (idx, value) in nums.iter().enumerate() {
            for i in 0..*value {
                if (i | (i + 1)) == *value {
                    result[idx] = i;
                    break;
                }
            }
        }

        result
    }
}

